src/components/UASTViewer.js

```diff
@@ -1,18 +1,35 @@
 import React, { Component } from 'react';
 import styled from 'styled-components';
+import Node from './uast/Node';
+import {eventHandler} from './uast/EventHandler';
+import {font, background} from '../styling/variables';
 
 const Container = styled.div`
+  position: absolute;
+  height: 100%;
+  width: 100%;
+  overflow: auto;
+  font-family: ${font.family.code};
+  font-size: ${font.size.large};
+  background: ${background.main};
+  line-height: ${font.lineHeight.large};
+  padding: 4px;
 `
 
 export default class UASTViewer extends Component {
+  constructor(props) {
+    super(props);
+    eventHandler.onNodeSelected = this.props.onNodeSelected;
```

Comment:
Having a singleton class that exists only to not pass around the `onNodeSelected` as prop does not help the quality (imho) or make the code easier to understand. And it does complicate testing a lot. Whereas passing the `onNodeSelected` as prop you can pass a `jest.fn()` to the component instead of having to set the callback on the singleton.

Also, selecting the nodes seems pretty core functionality of the component to me.